instrumentation/opentelemetry-instrumentation-fastapi/tests/test_fastapi_instrumentation.py

```diff
@@ -1024,13 +1028,65 @@ def client_response_hook(send_span, scope, message):
             )
 
 
+def get_distribution_with_fastapi(*args, **kwargs):
+    dist = args[0]
+    if dist == "fastapi~=0.58":
+        # Value does not matter. Only whether an exception is thrown
+        return None
+    raise DistributionNotFound()
+
+
+def get_distribution_without_fastapi(*args, **kwargs):
+    raise DistributionNotFound()
+
+
 class TestAutoInstrumentation(TestBaseAutoFastAPI):
     """Test the auto-instrumented variant
 
     Extending the manual instrumentation as most test cases apply
     to both.
     """
 
+    def test_entry_point_exists(self):
```

Comment:
Do you think it would be useful to extract this into a common test suite to test autoinstrumentation in general? Not blocking for this pr.

Reply:
I'll make an issue for this and try when I apply this to another instrumentation. Similar to the last comment, we would need to make sure that it would still fail with breaking changes.

Reply:
https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2864